comp/logs/agent/config/config_keys.go

```diff
@@ -110,23 +119,39 @@ func (l *LogsConfigKeys) devModeUseProto() bool {
 }
 
 func (l *LogsConfigKeys) compressionKind() string {
-	compressionKind := l.getConfig().GetString(l.getConfigKey("compression_kind"))
-	switch compressionKind {
-	case "zstd", "gzip":
-		log.Debugf("Logs agent is using: %s compression", compressionKind)
+	// Check if additional endpoints are configured
+	endpoints, _ := l.getAdditionalEndpoints()
+	if len(endpoints) > 0 {
+		log.Debugf("Additional endpoints detected, pipeline: %s falling back to gzip compression for compatibility", l.prefix)
+		return GzipCompressionKind
```

Comment:
I think we still want to use `compression_kind` here - but only if it's explicitly set (ex not the default). 
[You can do this with `IsConfigured`](https://github.com/DataDog/datadog-agent/blob/05e00bf3dd1da372b71d9d422056d09c43c572c4/pkg/config/model/types.go#L174). 

This way (for example in org2) we can still use zstd with dual shipping if we explicitly opt-in to it.

Reply:
So the control flow would be something like:

```
if IsConfigured("compression_kind"):
    return config.Get("compression_kind")
else:
    if using addtional_endpoints:
        return gzip
    else:
        return config.Get("compression_kind")
```

Reply:
Ideally let's also add a UT to cover this specific case.